api/specs.proto

```diff
@@ -274,6 +274,9 @@ message ClusterSpec {
 
 	// TaskDefaults specifies the default values to use for task creation.
 	TaskDefaults task_defaults = 7 [(gogoproto.nullable) = false];
+
+	// LockKey is the encryption key for the raft data
```

Comment:
> For now, as I understand, this is limited to raft. Do we have plans on opening this up to "at rest" data? Would it be more accurate to say this?

Sorry, yes, this is intended for everything at rest - I just hadn't implemented that part yet, and though I was going to do that in another PR, so I was intended on updating it then when the feature was in.  After some discussion with @aaronlehmann this is PR is going to implement *everything* nuts to bolts, so I'll update the comment.  :)

> Also, do we need to store the new and old key here in case the migration is interrupted?

I'm not sure the old key inside the `ClusterSpec` itself is necessary, since we don't need to be able to read old data except on starting up.  And on starting up, we won't be able to read data in here anyway.  :)  We just need the new key in order to write new data.

I think we are going to not use a shared key though, so I guess it won't be distributed via raft anymore.  If we have a data encryption key  + key encryption key, then yes if the data encryption key is rotated we'll have to keep the old one around.